### PR TITLE
fix: deprecate  sample_random_indices

### LINKED ISSUE #132: Remove sample_random_indices
**Describe the bug**
Currently we use sample_random_indices to sample random neuron weights. This feature is not very useful in practice and often introduces scope for multiple bugs. 

Let's remove that/move it to a separate branch

### PR DESCRIPTION
**Pull request by rheasukthanker**
#### Reference Issues/PRs
Closes #132 

#### What does this implement/fix? Explain your changes.
Deprecate  `sample_random_indices ` argument in forward()

### CODE REVIEW

File: whittle/models/gpt/blocks/mlp.py [hunk 1]
```diff
@@ -22,23 +22,16 @@ def __init__(self, config: Config) -> None:
         self.sub_network_intermediate_size = self.intermediate_size
 
     def set_sub_network(
-        self,
-        sub_network_n_embd: int,
-        sub_network_intermediate_size: int,
-        sample_random_indices: bool = False,
+        self, sub_network_n_embd: int, sub_network_intermediate_size: int
```
Comment (aaronkl) [inline whittle/models/gpt/blocks/mlp.py:25]: somehow the formatting seems to have changed? Could you re-run ruff?

**Review by aaronkl — APPROVED** [review]
LGTM. Could you maybe re-run ruff?

File: whittle/models/gpt/blocks/mlp.py (hunk 1 above)
Comment (rheasukthanker) [reply whittle/models/gpt/blocks/mlp.py:25]: Ruff doesn't reformat this somehow :(

Comment (rheasukthanker) [reply whittle/models/gpt/blocks/mlp.py:25]: I think the formatting changed, just because there are lesser arguments right now

**Pull request merged**